src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -351,13 +376,6 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		this._inputModel = model;
 
 		return new Promise<void>((resolve, reject) => {
-			// only create the provider maps first time the dialog gets called
-			if (this._providerTypes.length === 0) {
-				entries(this._capabilitiesService.providers).forEach(p => {
-					this._providerTypes.push(p[1].connection.displayName);
-					this._providerNameToDisplayNameMap[p[0]] = p[1].connection.displayName;
-				});
-			}
```

Comment:
Do we need to call `this.setConnectionProviders` from here? I see the new code calls it from the constructor instead of here but I'm not sure if there was some reason we were doing this logic from `showDialog` before

Reply:
with the reactive implementation we no longer need to update the providers before showing dialog.  
we will always have the up to date providers at this point.